### PR TITLE
Remove Duplicate Namespace Declarations

### PR DESCRIPTION
**Pull request by JeremyMarzano-ISPA**
### Description

- Some of our users have reported issues opening KML's that are generated with duplicate namespace declarations.
- This PR checks for a duplicate namespace declaration and deletes it.
- Duplicate namespaces can include xsi, gmd, etc.
- The PR adds a function that searches for namespace declarations starting with "xmlns:". It then checks for all duplicate namespace declarations until the end of the block ">". Any duplicate declarations are removed by setting the KML text to the slices before and after that namespace.
- Note: I work for Novetta, and have written this code for them.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
[Please sign the CLA before we review this PR.](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla)

Welcome to the Cesium community @JeremyMarzano-ISPA!

Can you please send in a [Contributor License Agreement](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) so that we can review and merge this pull request?

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by ggetz** [comment]
Hi @JeremyMarzano-ISPA, thanks for the PR. @tfili will give this a review shortly!

**Comment by tfili** [comment]
@JeremyMarzano-ISPA Thanks for your PR.

This will only detect duplicates if it is defined twice, but not 3 times. It will find occurrence `1`, then search for occurrence `2` and remove it. Then it will start its search after occurrence `1`, so when it find occurrence `3` it doesn't realize it already exists.

**Comment by JeremyMarzano-ISPA** [comment]
Thank you @ggetz. I have updated CONTRIBUTORS.md with my information.

@tfili Thank you for the review. I have updated the function to continue to search for duplicates of the same namespace, until no duplicates are found. I have also updated the test case to include this. Please let me know if you find any other issues, or have any additional comments. Thank you.

**Comment by hpinkos** [comment]
@JeremyMarzano-ISPA can you merge in master when you have a chance?

**Comment by JeremyMarzano-ISPA** [comment]
@hpinkos Thank you for letting me know. I have merged with master.

File: Source/DataSources/KmlDataSource.js [hunk 1]
```diff
@@ -2369,6 +2391,9 @@ define([
                             //Insert missing namespaces
                             text = insertNamespaces(text);
 
+                            //Remove Duplicate Namespaces
+                            text = removeDuplicateNamespaces(text);
```
Comment (tfili) [inline Source/DataSources/KmlDataSource.js]: This won't get called for kmz files. Look for all the occurrences of `insertNamespaces(...)`. This should be called in the same places.

File: Source/DataSources/KmlDataSource.js [hunk 2]
```diff
@@ -282,6 +282,28 @@ define([
...
+
+        while ((index !== -1) && (index < endDeclaration)) {
+            namespace = text.slice(index, text.indexOf('\"', index));
+            startIndex = index;
+            index = text.indexOf(namespace, index + 1);
+            if (index !== -1) {
+                endIndex = text.indexOf('\"', (text.indexOf('\"', index) + 1));
+                text = text.slice(0, index -1) + text.slice(endIndex + 1, text.length);
+                index = text.indexOf('xmlns:', startIndex - 1);
+            }
```
Comment (tfili) [inline Source/DataSources/KmlDataSource.js]: Else should be on same line as the previous curly brace.

File: Source/DataSources/KmlDataSource.js (hunk 1 above)
Comment (JeremyMarzano-ISPA) [reply Source/DataSources/KmlDataSource.js]: Thank you for finding that. I have updated the PR with a call for KMZ's as well. Please let me know if you find any other issues. Thank you.

**Comment by tfili** [comment]
Thanks @JeremyMarzano-ISPA.

**Pull request merged**